src/main.rs

```diff
@@ -287,11 +305,61 @@ async fn monitor(
             return Reconnect::NoUpstream;
         }
 
-        should_check_upstreams_latency += 1;
         tokio::time::sleep(tokio::time::Duration::from_millis(100)).await;
     }
 }
 
+fn check_update_proxy() {
+    info!("Checking for latest released version...");
+    // Determine the OS and map to the asset name
+    let os = std::env::consts::OS;
+    let target_asset = match os {
+        "linux" => "demand-cli-linux",
+        "macos" => "demand-cli-macos",
+        "windows" => "demand-cli-windows.exe",
+        _ => {
+            error!("Warning: Unsupported OS '{}', skipping update", os);
+            return;
+        }
+    };
+
+    debug!("Target-arch... {}", target_asset);
+    info!("Current version... {}", cargo_crate_version!());
+
+    let updater = match backends::github::Update::configure()
+        .repo_owner("demand-open-source")
+        .repo_name("demand-cli")
+        .bin_name("demand-cli")
```

Comment:
```suggestion
        .repo_owner(REPO_OWNER)
        .repo_name(REPO_NAME)
        .bin_name(BIN_NAME)
```

And they should be defined as `const REPO_OWNER: &str = ...`